implementations/lighthouse/README.md

```diff
@@ -0,0 +1,4771 @@
+  }
+}
+</code></pre>
+</details>
+</li>
+<li><code>556A</code> SHOULD use 400 status code on document parsing failure when accepting application/graphql-response+json
+<details>
+<summary>Response status code is not 400</summary>
+<pre><code class="lang-json">{
+  "statusText": "OK",
+  "status": 200,
+  "headers": {
+    "x-powered-by": "PHP/8.1.17",
+    "host": "localhost:8123",
+    "date": "<timestamp>",
+    "content-type": "application/json",
+    "connection": "close",
+    "cache-control": "no-cache, private"
+  },
+  "body": {
+    "errors": [
+      {
+        "message": "Syntax Error: Expected Name, found <EOF>",
+        "locations": [
+          {
+            "line": 1,
+            "column": 2
+          }
+        ],
+        "extensions": {
+          "line": 383,
+          "file": "/app/vendor/webonyx/graphql-php/src/Language/Parser.php"
+        }
+      }
+    ]
+  }
+}
+</code></pre>
+</details>
+</li>
+<li><code>51FE</code> SHOULD use 4xx or 5xx status codes on document validation failure when accepting application/graphql-response+json
+<details>
+<summary>Response status is not between 400 and 599</summary>
+<pre><code class="lang-json">{
+  "statusText": "OK",
+  "status": 200,
+  "headers": {
+    "x-powered-by": "PHP/8.1.17",
+    "host": "localhost:8123",
+    "date": "<timestamp>",
+    "content-type": "application/json",
+    "connection": "close",
+    "cache-control": "no-cache, private"
+  },
+  "body": {
+    "errors": [
+      {
+        "message": "Syntax Error: Expected Name, found Int \"8\"",
+        "locations": [
+          {
+            "line": 1,
+            "column": 3
+          }
+        ],
+        "extensions": {
+          "line": 383,
+          "file": "/app/vendor/webonyx/graphql-php/src/Language/Parser.php"
+        }
+      }
+    ]
+  }
+}
+</code></pre>
+</details>
+</li>
+<li><code>74FF</code> SHOULD use 400 status code on document validation failure when accepting application/graphql-response+json
+<details>
+<summary>Response status code is not 400</summary>
+<pre><code class="lang-json">{
+  "statusText": "OK",
+  "status": 200,
+  "headers": {
+    "x-powered-by": "PHP/8.1.17",
+    "host": "localhost:8123",
+    "date": "<timestamp>",
+    "content-type": "application/json",
+    "connection": "close",
+    "cache-control": "no-cache, private"
+  },
+  "body": {
+    "errors": [
+      {
+        "message": "Syntax Error: Expected Name, found Int \"8\"",
+        "locations": [
+          {
+            "line": 1,
+            "column": 3
+          }
+        ],
+        "extensions": {
+          "line": 383,
+          "file": "/app/vendor/webonyx/graphql-php/src/Language/Parser.php"
+        }
+      }
+    ]
+  }
+}
+</code></pre>
+</details>
+</li>
+</ol>
+
+<h2>Errors</h2>
+The server <b>MUST</b> support these.
+<ol>
+<li><code>7267</code> MUST require a request body on POST
+<details>
+<summary>Response body execution result does not have a property "errors"</summary>
```

Comment:
Why would `errors` be required here? In case of a missing response body, responding with status code 400 and a response format of the servers choosing seems fine. The graphql-over-http specification does not mandate the usage of the GraphQL response format for non-200 responses.

Reply:
Yes, this was a flawed assumption. There's a PR though: https://github.com/graphql/graphql-over-http/pull/241.

The reasoning is that if you're responding with a JSON, graphql-http assumes you're responding with an error GQL style - hence the `errors` field requirement.

Nevertheless, you're right: I'll be adapting the audits to reflect this.

Reply:
I don't see how https://github.com/graphql/graphql-over-http/pull/241 would change anything about this. What part am I missing?

Thanks for taking care of this though - I would also be happy to make a pull request if you can point me in the right direction.